auth.go

```diff
@@ -0,0 +1,99 @@
+package winrm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/masterzen/winrm/soap"
+	"launchpad.net/gwacl/fork/http"
+	"launchpad.net/gwacl/fork/tls"
+)
+
+func newCustomTransport(endpoint *Endpoint) (*http.Transport, error) {
+	var (
+		cert tls.Certificate
+		err  error
+	)
+
+	// test if it's valid key,cert client base
+	if isCertAndKey(endpoint.Cert, endpoint.Key) {
+		cert, err = tls.X509KeyPair(endpoint.Cert, endpoint.Key)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// make the custom transport
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: endpoint.Insecure,
+			Certificates:       []tls.Certificate{cert},
+		},
+		//ResponseHeaderTimeout: endpoint.Timeout,
+	}
+
+	if endpoint.CACert != nil && len(endpoint.CACert) > 0 {
+		certPool, err := readCACerts(endpoint.CACert)
+		if err != nil {
+			return nil, err
+		}
+
+		transport.TLSClientConfig.RootCAs = certPool
+	}
+
+	return transport, nil
+
+}
+
+type clientAuthRequest struct {
+	transport http.RoundTripper
+}
+
+// parse func reads the response body and return it as a string
+func parse(response *http.Response) (string, error) {
+
+	// if we recived the content we expected
+	if strings.Contains(response.Header.Get("Content-Type"), "application/soap+xml") {
+		body, err := ioutil.ReadAll(response.Body)
+		defer func() {
+			if err := response.Body.Close(); err != nil {
+				panic(err)
+			}
+		}()
+		if err != nil {
+			return "", fmt.Errorf("error while reading request body %s", err)
+		}
+
+		return string(body), nil
+	}
+
+	return "", fmt.Errorf("invalid content type")
+}
+
+func (c clientAuthRequest) Post(client *Client, request *soap.SoapMessage) (string, error) {
```

Comment:
There's a lot of similarities between this code and the code in `http.go`. Isn't there a way to not have duplicate code at all (or at least minimize it)?